bundles/org.openhab.binding.haassohnpelletstove/src/main/java/org/openhab/binding/haassohnpelletstove/internal/HaasSohnpelletstoveHandler.java

```diff
@@ -108,82 +108,70 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
-    /**
-     * Calls the service to update the oven data
-     *
-     * @param postdata
-     */
-    private boolean updateOvenData(@Nullable String postdata) {
-        Helper message = new Helper();
-        if (serviceCommunication.updateOvenData(postdata, message, this.getThing().getUID().toString())) {
-            updateStatus(ThingStatus.ONLINE);
-            return true;
-        } else {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    message.getStatusDesription());
-            return false;
-        }
-    }
-
     @Override
     public void initialize() {
         logger.debug("Initializing haassohnpelletstove handler for thing {}", getThing().getUID());
         config = getConfigAs(HaasSohnpelletstoveConfiguration.class);
-        boolean validConfig = true;
-        String errors = "";
-        String statusDescr = null;
-        if (config.refreshRate < 0 && config.refreshRate > 999) {
-            errors += " Parameter 'refresh Rate' greater then 0 and less then 1000.";
-            statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1000.";
-            validConfig = false;
+        if (config.refreshRate < 0 || config.refreshRate > 1000) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Parameter 'refresh Rate' msut be in the range 0-1000!");
+            return;
         }
-        if (config.hostIP == null) {
-            errors += " Parameter 'hostIP' must be configured.";
-            statusDescr = "IP Address must be configured!";
-            validConfig = false;
+        if (config.hostIP.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IP Address must be configured!");
+            return;
         }
-        if (config.hostPIN == null) {
-            errors += " Parameter 'hostPin' must be configured.";
-            statusDescr = "PIN must be configured!";
-            validConfig = false;
+
+        if (config.hostPIN.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Parameter 'hostPin' must be configured!");
+            return;
+        }
+
+        if (config.refreshRate < 1 || config.refreshRate > 1000000) {
```

Comment:
The refreshRate was already checked above, this seems redundant, ore the wrong property is checked.

Reply:
Correct, remaining from refactoring. Will correct